src/proxy_handler.cc

```diff
@@ -0,0 +1,160 @@
+			      portno = statement->tokens_[1];
+			      // verify that portno is a valid number
+			      for (size_t i = 0; i < portno.size(); i++){
+				if (!std::isdigit(portno[i])){
+				  return RequestHandler::invalid_config;
+				}
+			      }
+			    }
+			    else {
+			      return RequestHandler::invalid_config;
+			    }
+			  }
+    		// other statements
+  			}
+			// make some final checks:
+			if ((host_name == "") || (portno == "")) {
+			  return RequestHandler::invalid_config;
+			}
+  			return RequestHandler::ok;
+		}
+	  
+	  using boost::asio::ip::tcp;
+	  
+  		RequestHandler::Status ProxyHandler::HandleRequest(const Request& request, Response* response){
+  			boost::asio::io_service io_service;
+		    tcp::resolver resolver(io_service);
+		    tcp::resolver::query query(host_name, "http");
+		    tcp::resolver::iterator endpoint_iterator = resolver.resolve(query);
+
+		    tcp::socket socket(io_service);
+		    boost::asio::connect(socket, endpoint_iterator);
+		    
+		    std::string uri = request.uri() == uri_prefix ? "/" : request.uri();
+
+		    //int loc = uri.substr(1).find_first_of("/");
+
+		    int findLoc = uri.find(uri_prefix);
+
+		    if(findLoc != -1){
+		    	uri.erase(findLoc, uri_prefix.size());
+		    }
+
+		    std::cout << "URIBEINGSENT:  " << uri << std::endl;
+ 		    boost::asio::streambuf requestBuf;
+		    std::ostream request_stream(&requestBuf);
+		    request_stream << "GET " << uri << " HTTP/1.1\r\n";
+		    request_stream << "Host: " << host_name << ":" << portno << "\r\n"; 
+		    request_stream << "Accept: */*\r\n";
+		    request_stream << "Connection: close\r\n\r\n";
+
+		    // Send the request.
+		    boost::asio::write(socket, requestBuf);
+
+		    // Read the response status line. The response streambuf will automatically
+		    // grow to accommodate the entire line. The growth may be limited by passing
+		    // a maximum size to the streambuf constructor.
+		    boost::asio::streambuf resp;
+		    boost::asio::read_until(socket, resp, "\r\n");
+
+		    // Check that response is OK.
+		    std::istream response_stream(&resp);
+		    std::string http_version;
+		    response_stream >> http_version;
+		    std::cout << "http version: " << http_version << std::endl;
+
+		    unsigned int status_code;
+		    response_stream >> status_code;
+		    std::cout << "status code: " << status_code << std::endl;
+
+		    std::string status_message;
+		    std::getline(response_stream, status_message);
+		    std::cout << "status message: " << status_message << std::endl;
+		    
+		    if (!response_stream || http_version.substr(0, 5) != "HTTP/")
+		    {
+		      std::cout << "Invalid response\n";
+		      //return 1;
+		    }
+		    if (status_code == 200)
+		    {
+		      std::cout << "Response returned with status code " << status_code << "\n";
+		      response->SetStatus(Response::ok);
+		      //return 1;
+		    }
+		    else if(status_code == 302 || status_code == 301){
```

Comment:
Oh maybe I misunderstand this assignment. I thought the proxy should handle the 302.